OK, now this PR almost fully works for Musescore (at commit [c106c4fe18efeb32bdb9c46a6475ba530f769fe6](https://github.com/musescore/MuseScore/tree/c106c4fe18efeb32bdb9c46a6475ba530f769fe6)), when modified as follows:

```diff
diff --git i/CMakeLists.txt w/CMakeLists.txt
index 6754b1fc3e..2d453a62c2 100644
--- i/CMakeLists.txt
+++ w/CMakeLists.txt
@@ -153,6 +153,7 @@ option(MUE_COMPILE_USE_SYSTEM_FREETYPE "Try use system freetype" OFF)
 option(MUE_COMPILE_USE_SYSTEM_HARFBUZZ "Try use system harfbuzz" OFF)
 option(MUE_COMPILE_USE_SYSTEM_OPUS "Try use system opus" OFF)
 option(MUE_COMPILE_USE_SYSTEM_OPUSENC "Try use system libopusenc" OFF)
+option(MUE_COMPILE_USE_SYSTEM_MNXDOM "Try use system mnxdom" OFF)
 
 # === Debug ===
 option(MUE_ENABLE_LOAD_QML_FROM_SOURCE "Load qml files from source (not resource)" OFF)
diff --git i/src/importexport/mnx/CMakeLists.txt w/src/importexport/mnx/CMakeLists.txt
index 6e4a6fa73b..9c62fcfe4e 100644
--- i/src/importexport/mnx/CMakeLists.txt
+++ w/src/importexport/mnx/CMakeLists.txt
@@ -9,13 +9,22 @@ muse_create_module(iex_mnx)
 include(FetchContent)
 
 set(mnxdom_BUILD_TESTING OFF CACHE BOOL "Do not build tests for mnx")
-FetchContent_Declare(
-    mnxdom
-    GIT_REPOSITORY https://github.com/rpatters1/mnxdom.git
-    GIT_TAG        f471206335392ff52f49dbb61284cd527c594672
-)
-
-FetchContent_MakeAvailable(mnxdom)
+if(MUE_COMPILE_USE_SYSTEM_MNXDOM)
+    find_package(PkgConfig REQUIRED)
+    pkg_check_modules(mnxdom REQUIRED IMPORTED_TARGET mnxdom)
+    set(MNXDOM_LIB PkgConfig::mnxdom)
+    target_compile_definitions(iex_mnx PRIVATE MNXDOM_SYSTEM)
+    message(STATUS "Using system-installed mnxdom")
+else()
+    FetchContent_Declare(
+        mnxdom
+        GIT_REPOSITORY https://github.com/rpatters1/mnxdom.git
+        GIT_TAG        f471206335392ff52f49dbb61284cd527c594672
+    )
+    FetchContent_MakeAvailable(mnxdom)
+    set(MNXDOM_LIB mnxdom)
+    message(STATUS "Using fetched mnxdom")
+endif()
 
 target_sources(iex_mnx PRIVATE
     mnxmodule.cpp
@@ -42,7 +51,7 @@ target_sources(iex_mnx PRIVATE
 )
 
 target_link_libraries(iex_mnx PRIVATE
-    mnxdom
+    ${MNXDOM_LIB}
     engraving
 )
 
diff --git i/src/importexport/mnx/internal/export/mnxexporter.h w/src/importexport/mnx/internal/export/mnxexporter.h
index e6cbc41d11..8c95b6eea2 100644
--- i/src/importexport/mnx/internal/export/mnxexporter.h
+++ w/src/importexport/mnx/internal/export/mnxexporter.h
@@ -25,7 +25,11 @@
 #include <unordered_set>
 #include <vector>
 
+#ifdef MNXDOM_SYSTEM
+#include <mnxdom/mnxdom.h>
+#else
 #include "mnxdom.h"
+#endif
 #include "types/ret.h"
 
 #include "engraving/types/types.h"
diff --git i/src/importexport/mnx/internal/import/mnximporter.h w/src/importexport/mnx/internal/import/mnximporter.h
index dea52be30d..f857dfd942 100644
--- i/src/importexport/mnx/internal/import/mnximporter.h
+++ w/src/importexport/mnx/internal/import/mnximporter.h
@@ -28,7 +28,11 @@
 #include "engraving/types/propertyvalue.h"
 #include "engraving/types/types.h"
 
+#ifdef MNXDOM_SYSTEM
+#include <mnxdom/mnxdom.h>
+#else
 #include "mnxdom.h"
+#endif
 
 namespace mu::engraving {
 class Chord;
diff --git i/src/importexport/mnx/internal/import/mnximportmarkings.cpp w/src/importexport/mnx/internal/import/mnximportmarkings.cpp
index ff310e619f..6e34ddca11 100644
--- i/src/importexport/mnx/internal/import/mnximportmarkings.cpp
+++ w/src/importexport/mnx/internal/import/mnximportmarkings.cpp
@@ -36,7 +36,11 @@
 #include "engraving/types/symid.h"
 #include "log.h"
 
+#ifdef MNXDOM_SYSTEM
+#include <mnxdom/mnxdom.h>
+#else
 #include "mnxdom.h"
+#endif
 
 using namespace mu::engraving;
 
diff --git i/src/importexport/mnx/internal/import/mnximportpartmeasures.cpp w/src/importexport/mnx/internal/import/mnximportpartmeasures.cpp
index 17a5710801..1c727e5d9b 100644
--- i/src/importexport/mnx/internal/import/mnximportpartmeasures.cpp
+++ w/src/importexport/mnx/internal/import/mnximportpartmeasures.cpp
@@ -55,7 +55,11 @@
 #include "engraving/dom/tuplet.h"
 #include "engraving/dom/volta.h"
 
+#ifdef MNXDOM_SYSTEM
+#include <mnxdom/mnxdom.h>
+#else
 #include "mnxdom.h"
+#endif
 
 using namespace mu::engraving;
 
diff --git i/src/importexport/mnx/internal/notationmnxreader.cpp w/src/importexport/mnx/internal/notationmnxreader.cpp
index 3ffab4ba71..cd2f08fafa 100644
--- i/src/importexport/mnx/internal/notationmnxreader.cpp
+++ w/src/importexport/mnx/internal/notationmnxreader.cpp
@@ -29,7 +29,11 @@
 #include "io/file.h"
 #include "log.h"
 
+#ifdef MNXDOM_SYSTEM
+#include <mnxdom/mnxdom.h>
+#else
 #include "mnxdom.h"
+#endif
 
 using namespace mu::iex::mnxio;
 using namespace mu::engraving;
diff --git i/src/importexport/mnx/internal/shared/mnxtypesconv.h w/src/importexport/mnx/internal/shared/mnxtypesconv.h
index 3a644708ff..2b62d94fe3 100644
--- i/src/importexport/mnx/internal/shared/mnxtypesconv.h
+++ w/src/importexport/mnx/internal/shared/mnxtypesconv.h
@@ -28,7 +28,11 @@
 #include "engraving/dom/tuplet.h"
 #include "engraving/types/types.h"
 
+#ifdef MNXDOM_SYSTEM
+#include <mnxdom/mnxdom.h>
+#else
 #include "mnxdom.h"
+#endif
 
 namespace mu::engraving {
 struct NoteVal;
diff --git i/src/importexport/mnx/tests/mnx_tests.cpp w/src/importexport/mnx/tests/mnx_tests.cpp
index 4b5b94a6ea..529c2878dd 100644
--- i/src/importexport/mnx/tests/mnx_tests.cpp
+++ w/src/importexport/mnx/tests/mnx_tests.cpp
@@ -54,7 +54,11 @@
 #include "types/ret.h"
 #include "engraving/rw/rwregister.h"
 
+#ifdef MNXDOM_SYSTEM
+#include <mnxdom/mnxdom.h>
+#else
 #include "mnxdom.h"
+#endif
 
 using namespace mu::engraving;
 using namespace mu::iex::mnxio;
```

However, it fails with:

```
FAILED: [code=1] src/importexport/mnx/CMakeFiles/iex_mnx.dir/Unity/unity_1_cxx.cxx.o 
/nix/store/a245z3cvf9x9sn0xlk6k8j9xhxbhda1z-gcc-wrapper-15.2.0/bin/g++ -DKORS_LOGGER_QT_SUPPORT -DKORS_PROFILER_ENABLED -DMNXDOM_SYSTEM -DNDEBUG -DQT_CORE_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_DEBUG -DQT_OPENGL_LIB -DQT_QMLINTEGRATION_LIB -DQT_QML_LIB -DQT_QUICK_LIB -DQT_WIDGETS_LIB -I/build/source/build/src/importexport/mnx -I/build/source/src/importexport/mnx -I/build/source/build/src/importexport/mnx/iex_mnx_autogen/include -I/build/source/build -I/build/source/src -I/build/source -I/build/source/framework -I/build/source/framework/global -I/build/source/framework/testing/thirdparty/googletest/googletest/include -I/build/source/src/framework -I/build/source/src/framework/global -I/build/source/src/framework/testing/thirdparty/googletest/googletest/include -isystem /nix/store/kavwy4v9p1jhf952dzipkjj5l1hjx110-qtbase-6.10.1/include/QtCore -isystem /nix/store/kavwy4v9p1jhf952dzipkjj5l1hjx110-qtbase-6.10.1/mkspecs/linux-g++ -isystem /nix/store/kavwy4v9p1jhf952dzipkjj5l1hjx110-qtbase-6.10.1/include/QtGui -isystem /nix/store/pma97xn76wlzmsqj018bl3df6mywgfl7-qtdeclarative-6.10.1/include/QtQuick -isystem /nix/store/pma97xn76wlzmsqj018bl3df6mywgfl7-qtdeclarative-6.10.1/include/QtQml -isystem /nix/store/pma97xn76wlzmsqj018bl3df6mywgfl7-qtdeclarative-6.10.1/include/QtQmlIntegration -isystem /nix/store/kavwy4v9p1jhf952dzipkjj5l1hjx110-qtbase-6.10.1/include/QtNetwork -isystem /nix/store/kavwy4v9p1jhf952dzipkjj5l1hjx110-qtbase-6.10.1/include/QtOpenGL -isystem /nix/store/kavwy4v9p1jhf952dzipkjj5l1hjx110-qtbase-6.10.1/include/QtWidgets -O3 -DNDEBUG -std=gnu++17 -Wall -Wextra -fPIC -Winvalid-pch -include /build/source/build/src/framework/pch/CMakeFiles/muse_pch.dir/cmake_pch.hxx -MD -MT src/importexport/mnx/CMakeFiles/iex_mnx.dir/Unity/unity_1_cxx.cxx.o -MF src/importexport/mnx/CMakeFiles/iex_mnx.dir/Unity/unity_1_cxx.cxx.o.d -o src/importexport/mnx/CMakeFiles/iex_mnx.dir/Unity/unity_1_cxx.cxx.o -c /build/source/build/src/importexport/mnx/CMakeFiles/iex_mnx.dir/Unity/unity_1_cxx.cxx
In file included from /nix/store/1qskn8pmja5kwz2pdi319c1wxxbldfsz-mnxdom-2.0-unstable-2026-02-12-dev/include/mnxdom/mnxdom.h:62,
                 from /build/source/src/importexport/mnx/internal/import/mnximporter.h:32,
                 from /build/source/src/importexport/mnx/internal/notationmnxreader.cpp:23,
                 from /build/source/build/src/importexport/mnx/CMakeFiles/iex_mnx.dir/Unity/unity_1_cxx.cxx:4:
/nix/store/1qskn8pmja5kwz2pdi319c1wxxbldfsz-mnxdom-2.0-unstable-2026-02-12-dev/include/mnxdom/BaseTypes.h:1021:13: error: redefinition of 'template<class BasicJsonType, class EnumType, typename std::enable_if<std::is_enum<_Tp2>::value, int>::type <anonymous> > void nlohmann::json_abi_v3_12_0::detail::from_json(const BasicJsonType&, EnumType&)'
 1021 | inline void from_json(const BasicJsonType& j, EnumType& value)
      |             ^~~~~~~~~
In file included from /nix/store/6zpgsxs888i5ifqkj4bbb1czf2my655n-nlohmann_json-3.12.0/include/nlohmann/adl_serializer.hpp:14,
                 from /nix/store/6zpgsxs888i5ifqkj4bbb1czf2my655n-nlohmann_json-3.12.0/include/nlohmann/json.hpp:34,
                 from /nix/store/1qskn8pmja5kwz2pdi319c1wxxbldfsz-mnxdom-2.0-unstable-2026-02-12-dev/include/mnxdom/BaseTypes.h:36:
/nix/store/6zpgsxs888i5ifqkj4bbb1czf2my655n-nlohmann_json-3.12.0/include/nlohmann/detail/conversions/from_json.hpp:162:13: note: 'template<class BasicJsonType, class EnumType, typename std::enable_if<std::is_enum<_Tp2>::value, int>::type <anonymous> > void nlohmann::json_abi_v3_12_0::detail::from_json(const BasicJsonType&, EnumType&)' previously declared here
  162 | inline void from_json(const BasicJsonType& j, EnumType& e)
      |             ^~~~~~~~~
/nix/store/1qskn8pmja5kwz2pdi319c1wxxbldfsz-mnxdom-2.0-unstable-2026-02-12-dev/include/mnxdom/BaseTypes.h:1036:13: error: redefinition of 'template<class BasicJsonType, class EnumType, typename std::enable_if<std::is_enum<_Tp2>::value, int>::type <anonymous> > void nlohmann::json_abi_v3_12_0::detail::to_json(BasicJsonType&, EnumType)'
 1036 | inline void to_json(BasicJsonType& j, EnumType value) noexcept
      |             ^~~~~~~
In file included from /nix/store/6zpgsxs888i5ifqkj4bbb1czf2my655n-nlohmann_json-3.12.0/include/nlohmann/adl_serializer.hpp:15:
/nix/store/6zpgsxs888i5ifqkj4bbb1czf2my655n-nlohmann_json-3.12.0/include/nlohmann/detail/conversions/to_json.hpp:341:13: note: 'template<class BasicJsonType, class EnumType, typename std::enable_if<std::is_enum<_Tp2>::value, int>::type <anonymous> > void nlohmann::json_abi_v3_12_0::detail::to_json(BasicJsonType&, EnumType)' previously declared here
  341 | inline void to_json(BasicJsonType& j, EnumType e) noexcept
      |             ^~~~~~~
```

Which seems related to what's happening here:

https://github.com/rpatters1/mnxdom/blob/0c271fcb2bbe348b2ff372535996a7b882300915/src/BaseTypes.h#L975-L1049

What do you think @rpatters1 ?

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -79,6 +79,7 @@ endif()
 # Handle json_schema_validator: use system version or fetch
 if(USE_SYSTEM_JSON_SCHEMA_VALIDATOR)
     find_package(nlohmann_json_schema_validator REQUIRED CONFIG)
+    target_compile_definitions(mnxdom PRIVATE JSON_SCHEMA_VALIDATOR_SYSTEM)
     message(STATUS "Using system-installed nlohmann_json_schema_validator")
 else()
     # Disable building tests for json_schema_validator

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -60,6 +60,7 @@ set(NLOHMANN_JSON_VERSION 3.12.0)
 # Handle nlohmann_json: use system version or fetch
 if(USE_SYSTEM_NLOHMANN_JSON)
     find_package(nlohmann_json ${NLOHMANN_JSON_VERSION} REQUIRED CONFIG)
+    target_compile_definitions(mnxdom PRIVATE NLOHMANN_JSON_SYSTEM)
     message(STATUS "Using system-installed nlohmann_json")
 else()
     # Disable building tests for nlohmann_json

--- a/README.md
+++ b/README.md
@@ -41,6 +41,8 @@ Note that this code never uses references. Since every MNX class in the model is
 
 ### Setup Instructions
 
+#### Via `FetchContent`
+
 Include the top header in your source file.
 
 ```cpp
@@ -61,6 +63,27 @@ FetchContent_MakeAvailable(musx)
 
 # Also add somewhere:
 target_link_libraries(project PRIVATE mnxdom) # replace "project" with your actual project name
+```
+
+#### Via a system installation
+
+Install the library using `sudo cmake --install`. This should install:
+
+- Headers in `/usr/local/include/mnxdom`
+- `pkg-config` file in `/usr/local/lib/pkgconfig/mnxdom.pc`
+
+If you use `cmake` in your project, use:
+
+```cmake
+  find_package(PkgConfig REQUIRED)
+  pkg_check_modules(mnxdom REQUIRED IMPORTED_TARGET mnxdom)
+  target_link_libraries(project PRIVATE PkgConfig::mnxdom)
+```
+
+And include the top header in your source file with:
+
+```cpp
+#include <mnxdom/mnxdom.h>
 ```
 
 ### To Run the Tests

--- a/src/BaseTypes.h
+++ b/src/BaseTypes.h
@@ -30,7 +30,11 @@
 #include <type_traits>
 #include <utility>
 
+#ifdef NLOHMANN_JSON_SYSTEM
+#include <nlohmann/json.hpp>
+#else
 #include "nlohmann/json.hpp"
+#endif
 #include "BoilerplateMacros.h"
 
 /**

--- a/src/validation/SchemaValidate.cpp
+++ b/src/validation/SchemaValidate.cpp
@@ -21,7 +21,11 @@
  */
 #include "mnxdom.h"
 
+#ifdef JSON_SCHEMA_VALIDATOR_SYSTEM
+#include <nlohmann/json-schema.hpp>
+#else
 #include "nlohmann/json-schema.hpp"
+#endif
 #include "mnx_schema.xxd"
 
 namespace mnx {

